### PR TITLE
Optimize deploy

### PR DESCRIPTION
**Pull request by purplecabbage**
## Description

These changes use 2 lists to determine if things have changed since last build/deploy
- A build list is kept which records a hash of an action's src directory
- A deploy list records what was deployed, including a hash of action descriptor so that even changes to action inputs, or .env values will force a re-deploy.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by moritzraho — APPROVED** [review]

File: src/deploy-actions.js [hunk 1]
```diff
@@ -64,6 +66,8 @@ async function deployActions (config, deployConfig = {}, logFunc) {
     aioLogger.debug('Trimming out the manifest\'s actions...')
     filterEntities = undefined
     const builtActions = []
+    // this is a little weird, we are getting the list of built actions from the dist folder
```
Comment (shazron) [inline src/deploy-actions.js:69]: shall we log an issue for this tech debt?

File: src/deploy-actions.js [hunk 2]
```diff
@@ -160,6 +165,15 @@ async function deployWsk (scriptConfig, manifestContent, logFunc, filterEntities
     namespace: scriptConfig.ow.namespace
   }
 
+  const lastDeployedActionsPath = path.join(scriptConfig.root, 'dist', 'last-deployed-actions.json')
+  let lastDeployData = {}
+  if (fs.existsSync(lastDeployedActionsPath)) {
+    lastDeployData = await fs.readJson(lastDeployedActionsPath)
+  } else {
```
Comment (shazron) [inline src/deploy-actions.js:172]: remove unused else

**Review by shazron — APPROVED** [review]

File: src/deploy-actions.js (hunk 2 above)
Comment (purplecabbage) [reply src/deploy-actions.js:179]: done, I chose to output something in this case so users can know it is happening.

File: src/deploy-actions.js [hunk 3]
```diff
@@ -160,6 +165,15 @@ async function deployWsk (scriptConfig, manifestContent, logFunc, filterEntities
     namespace: scriptConfig.ow.namespace
   }
 
+  const lastDeployedActionsPath = path.join(scriptConfig.root, 'dist', 'last-deployed-actions.json')
```
[the remark this answers is not in the document]
Comment (purplecabbage) [reply src/deploy-actions.js:173]: They are each only used in 1 place, so I think it is okay as is.  If multiple things need to access this file by name then a constant would make sense.  As it is, its just direct.
I DO however think that the config should define the dist folder, and not just dist/<ext>/actions/

File: src/deploy-actions.js (hunk 1 above)
Comment (purplecabbage) [reply src/deploy-actions.js:71]: https://github.com/adobe/aio-lib-runtime/issues/194

**Review by MichaelGoberling — APPROVED** [review]

**Pull request merged**